core/main_loop.py

```diff
@@ -31,6 +31,26 @@
   #from osh.cmd_exec import Executor
 
 
+def _ExecutePromptCommand(ex):
+  # type: (Any) -> None
+  command = None
+  prompt_var = ex.mem.GetVar('PROMPT_COMMAND')
+  if prompt_var.tag == value_e.Undef:
+    return
+  elif prompt_var.tag == value_e.AssocArray:
+    command = prompt_var.d['0']
```

Comment:
This is what bash does, but OSH explicitly disallows it now.  That is `$a` is not equivalent to `${a[0]}` (I made this an error in the last few days, but it always behaved differently than bash.)

I would just `return` unless it's a string.